src/main/java/org/mule/extension/http/internal/request/HttpPollingSource.java

```diff
@@ -186,8 +209,50 @@ public void poll(PollContext<String, HttpResponseAttributes> pollContext) {
   }
 
   @Override
-  public void onRejectedItem(Result<String, HttpResponseAttributes> result, SourceCallbackContext sourceCallbackContext) {
-    LOGGER.debug("Item rejected by HTTP Polling Source in flow '{}' with result: '{}", location.getRootContainerName(),
+  public void onRejectedItem(Result<String, HttpResponseAttributes> result,
+                             SourceCallbackContext sourceCallbackContext) {
+    LOGGER.debug("Item rejected by HTTP Polling Source in flow '{}', result: '{}'", location.getRootContainerName(),
                  result.getOutput());
   }
+
+  private BindingContext buildContext(TypedValue<String> payload, HttpResponseAttributes attributes,
+                                      TypedValue<CursorStreamProvider> item) {
+    BindingContext.Builder builder = BindingContext.builder().addBinding("payload", payload);
+    builder.addBinding("attributes", TypedValue.of(attributes));
+    if (item != null) {
+      builder.addBinding("item", item);
+    }
+
+    return builder.build();
+  }
+
+  private static TypedValue<String> toTypedValue(String value, MediaType mediaType, Charset encoding) {
+    if (mediaType.equals(TEXT.withCharset(mediaType.getCharset().orElse(null)))) {
```

Comment:
null or some default charset?

Reply:
null, because if the mediatype is not coming with any charset (we are in the "orElse"), then it would fail the comparison if I set anything different